js/project-page-functions.js

```diff
@@ -0,0 +1,227 @@
+  /* smooth scrolling to sections with room for navbar */
+  var $rightSidebarNav = $("#right-sidebar-nav");
+  $rightSidebarNav.find("li a[href^='#']").on('click', function (e) {
+    // prevent default anchor click behavior
+    e.preventDefault();
+    // store hash
+    var hash = this.hash;
+    // animate
+    $('html, body').animate({
+      scrollTop: $(hash).offset().top - navHeight
+    }, 300, function () {
+      // when done, add hash to url
+      // (default click behaviour)
+      window.location.hash = hash;
+    });
+  });
+
+  /* Scroll to current section if URL has hash */
+  if (window.location.hash) {
+    $rightSidebarNav.find("li a[href='#" + window.location.hash + "']").trigger('click');
+  }
+
+  $(window).on('scroll resize', function () {
+    $('#sidebar-nav, #revisions-div').css('bottom', getVisibleHeight('footer'));
+  });
+}
+
+function getVisibleHeight(selector) {
+  var $el = $(selector),
+    scrollTop = $(this).scrollTop(),
+    scrollBot = scrollTop + $(this).height(),
+    elTop = $el.offset().top,
+    elBottom = elTop + $el.outerHeight(),
+    visibleTop = elTop < scrollTop ? scrollTop : elTop,
+    visibleBottom = elBottom > scrollBot ? scrollBot : elBottom;
+  if ((visibleBottom - visibleTop) > 0)
+    return visibleBottom - visibleTop;
+  else
+    return 0;
+}
+
+function showTenMore(){
+
+  var $revisions = $('#left-sidebar').find('#revisions');
+  var counter = 0;
+
+  // get the rows still hidden
+  var hiddenRows = $revisions.find('tr').filter(function() {
+    var $this = $(this);
+    return $this.css('display') === 'none';
+  });
+
+  // show the next batch of rows
+  while (counter < 10 && counter < hiddenRows.length) {
+    hiddenRows[counter].style.display = '';
+    counter++;
+  }
+
+  // if all rows are now visible, hide the View More link
+  if (counter >= hiddenRows.length) {
+    $revisions.find('#view_more_tr').css('display', 'none');
+  }
+}
+
+function printAll(){
+  var id = myOwner+"/"+myRepoName+"/"+myCommitId;
+  var api_domain = "api.door43.org";
+  var api_prefix = "";
+  switch(window.location.hostname){
+    case "dev.door43.org":
+      api_prefix = "dev-";
+      break;
+    case "test-door43.org.s3-website-us-west-2.amazonaws.com":
+      api_prefix = "test-";
+      break;
+  }
+  window.open("https://"+api_prefix+api_domain+"/tx/print?id="+id,'_blank');
+}
+
+/**
+ * Gets the link back to DCS for the current page
+ * @param pathname The value from window.location.pathname
+ */
+function getDcsLink(pathname) {
+
+  // split the URL into its parts
+  var path_parts = pathname.split('/');
+  path_parts = path_parts.filter(function(s) {
+    return s !== '';
+  });
+
+  // length must be at least 3
+  if (path_parts.length < 3)
+    return '';
+
+  // the first part should be 'u'
+  if (path_parts[0] !== 'u')
+    return '';
+
+  // the second and third parts are what we need
+  return 'https://git.door43.org/' + path_parts[1] + '/' + path_parts[2];
```

Comment:
This should all be done simply with: `return 'https://git.door43.org/'+myOwner+'/'+myRepoName;` which gets retrieved from the build_log.log, rather than relying on the URL being a certain format. This is done the same with the print URL.